### DISCUSSION TITLE
JSON parsing way to difficult

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/runtime#95463] [General] [answered]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/JSON-parsing-way-to-difficult/10521939)._

---
[severity:It's more difficult to complete my work]
It's way too hard to parse a json from the web. You need to create your own class.?????? What the heck. It's just way too complex for such a simple task. Look at how Qt has done it. Simple, reliable and easy to use. https://doc.qt.io/qt-6/qjsondocument.html

With the isArray(), isObject(), isBool(), isDouble(), osObject, isString and isUndefined() and the corresponding to...() functions it's very easy to use. And here in C#, creating a custom class just for JSON parsing? Far too complex.

---
#### Feedback Bot on 11/22/2023, 02:51 AM: 

(private comment, text removed)
#### Feedback Bot on 11/22/2023, 03:52 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by ghost** [comment]
Tagging subscribers to this area: @dotnet/area-system-text-json, @gregsdennis
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.
<details>
<summary>Issue Details</summary>
<hr />

_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/JSON-parsing-way-to-difficult/10521939)._

---
[severity:It's more difficult to complete my work]
It's way too hard to parse a json from the web. You need to create your own class.?????? What the heck. It's just way too complex for such a simple task. Look at how Qt has done it. Simple, reliable and easy to use. https://doc.qt.io/qt-6/qjsondocument.html

With the isArray(), isObject(), isBool(), isDouble(), osObject, isString and isUndefined() and the corresponding to...() functions it's very easy to use. And here in C#, creating a custom class just for JSON parsing? Far too complex.

---
### Original Comments

#### Feedback Bot on 11/22/2023, 02:51 AM: 

(private comment, text removed)
#### Feedback Bot on 11/22/2023, 03:52 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

<table>
  <tr>
    <th align="left">Author:</th>
    <td>vsfeedback</td>
  </tr>
  <tr>
    <th align="left">Assignees:</th>
    <td>-</td>
  </tr>
  <tr>
    <th align="left">Labels:</th>
    <td>

`area-System.Text.Json`

</td>
  </tr>
  <tr>
    <th align="left">Milestone:</th>
    <td>-</td>
  </tr>
</table>
</details>

**Comment by huoyaoyuan** [comment, marked as answer]
There are `JsonNode` and `JsonDocument` in recent versions of System.Text.Json. They can be used with `JsonSerializer`, or parsed directly with `JsonNode.Parse` and `JsonDocument.Parse`.
